docs/global/modules/core/pages/howto/modelql.adoc

```diff
@@ -0,0 +1,190 @@
+= ModelQL
+
+== Independent ModelQLClient
+
+ModelQL defines its own HTTP endpoint and provides server/client implementations for it.
+The `model-server` and the `mps-model-server-plugin` already implement this endpoint.
+The client can be created like this:
+
+[source,kotlin]
+--
+val client = ModelQLClient.builder().url("http://localhost/query").httpClient(httpClient).build()
+val result: List<String?> = client.query { root ->
+    root.children("modules").property("name").toList()
+}
+--
+
+== Integration with LightModelClient
+
+When creating a `LightModelClient` you can optionally provide a `ModelQLClient` instance,
+which allows to invoke `.query { ... }` (see below) on a node returned by the `LightModelClient`.
+
+[source,kotlin]
+--
+val modelqlClient = ModelQLClient.builder().build()
+val client = LightModelClient.builder().modelQLClient(modelqlClient).build()
+val result: List<String?> = client.getRootNode()!!.query {
+    it.children("modules").property("name").toList()
+}
+--
+
+== Type safe ModelQL API
+
+You can use the `model-api-gen-gradle` plugin to generate type safe extensions from your meta-model.
+Specify the `modelqlKotlinDir` property to enable the generation.
```

Comment:
Can you make this property to a link to the reference documentation for the plugin?